tutorials/performance/pipeline_compilations.rst

```diff
@@ -208,3 +208,54 @@ you can attach the effect as a child of the player as an invisible node. Make
 sure to disable the script attached to the hidden node or to hide any other
 nodes that could cause issues, which can be done by enabling **Editable
 Children** on the node.
+
+.. _doc_pipeline_compilations_shader_baker:
+
+Shader baker
+------------
+
+Since Godot 4.5, you can choose to bake shaders on export to improve initial
+startup time. This will generally not resolve existing stutters, but it will
+reduce the time it takes to load the game for the first time. This is especially
+the case when using Direct3D 12 or Metal, which have significantly slower initial
+shader compilation times than Vulkan due to the conversion step required.
+Godot's own shaders use GLSL and SPIR-V, but Direct3D 12 and Metal use
+different formats.
+
+.. note::
+
+    The shader baker can only bake the source into the intermediate format
+    (SPIR-V for Vulkan, DXIL for Direct3D 12, MIL for Metal). It cannot bake
+    the intermediate format into the final pipeline, as this is
+    dependent on the GPU driver and the hardware.
+
+    The shader baker is not a replacement for pipeline precompilation,
+    but it aims to complement it.
+
+When enabled, the shader baker will bundle compiled shader code into the PCK,
+which results in the shader compilation step being skipped entirely.
+The downside is that exporting will take slightly more time. The PCK file
+will be larger by a few megabytes.
+
+The shader baker is disabled by default, but you can enable it in each
+export preset in the Export dialog by ticking the :ui:`Shader Baker > Enabled`
+export option.
+
+Note that shader baking will only work for the same
+rendering driver that is currently being used in the editor. For example,
```

Comment:
This is not necessarily true, DXIL could be produced at some point from a Linux build (considering DXC does run on Linux and can produce DXIL) and the Windows build can currently produce Vulkan shaders even if it's currently using D3D12. The actual reality of it might be harder to explain across in this document though. I'd probably just rephrase it to say that exporting to certain drivers depends on the host platform rather than the current rendering driver.

Reply:
Done. I've tried to clarify it by mentioning which drivers each platform can export shaders for, although I didn't mention future features like D3D12 shader baking on Linux.